akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2AlpnSupport.scala

```diff
@@ -36,7 +37,16 @@ private[http] object Http2AlpnSupport {
 
   def isAlpnSupportedByJDK: Boolean =
     // ALPN is supported starting with JDK 9
-    JavaVersion.majorVersion >= 9
+    JavaVersion.majorVersion >= 9 ||
+      (classOf[SSLEngine].getMethods.exists(_.getName == "setHandshakeApplicationProtocolSelector")
+        && {
+          // This method only exists in the jetty-alpn provided implementation. If it exists an old version of the jetty-alpn-agent is active which is not supported
+          // on JDK>= 8u252. When running on such a JVM, you can either just remove the agent or (if you want to support older JVMs with the same command line),
+          // use jetty-alpn-agent >= 2.0.10
+          val jettyAlpnClassesAvailable = Try(Class.forName("sun.security.ssl.ALPNExtension")).toOption.exists(_.getDeclaredMethods.exists(_.getName == "init"))
```

Comment:
Tested and works.

(I got into a situation where with JDK 252 the correct agent was loaded, but an older agent was also on the classpath, which confusingly but correctly triggered this error...)